go/enclave/db/storage.go

```diff
@@ -46,6 +46,10 @@ func NewStorage(backingDB sql.EnclaveDB, chainConfig *params.ChainConfig, logger
 	}
 }
 
+func (s *storageImpl) Close() error {
+	return s.db.GetSQLDB().Close()
```

Comment:
Feels a bit weird that the storage knows that it needs to reach into its db, get the underlying db and close it, maybe worth having Close on the 'db' object?

Reply:
the top-level enclave only knows about the `Storage` interface.
The Storage produces stateDbs or whatever else is needed. I think this is fine as it gives us flexibility.